fms_extras/models/speculator.py

```diff
@@ -32,15 +32,34 @@ class MLPSpeculator(nn.Module):
         Number of entries in the tokenizer associated with the base model.
     n_predict : int
         Number of heads / number of tokens to guess ahead. Model size and speed scale with this value.
+    tie_emb : bool
+        If true, use a single set of embedding weights for every model head/stage
+    tie_head : bool
+        If true, use a single set of prediction weights for every model head/stage
+    tie_transitions : bool
+        If true, use a single set of internal projection weights for every model head/stage after the first.
+        The initial projection from the base model may have a different size, so that stays separate.
     """
 
-    def __init__(self, emb_dim=4096, inner_dim=0, vocab_size=32000, n_predict=3):
+    def __init__(
+        self,
+        emb_dim=4096,
+        inner_dim=0,
+        vocab_size=32000,
+        n_predict=3,
+        tie_emb=False,
+        tie_head=False,
+        tie_transition=False,
+        tie_wts=False,
```

Comment:
Mentioned this in an earlier PR, but I think if this is just used to set all to True, it would make more sense to just remove this and explicitly set the other tie parameters to True.